packages/react/src/components/AccountSettings/DeleteUser/types.ts

```diff
@@ -0,0 +1,23 @@
+import { AmplifyUser } from '@aws-amplify/ui';
+
+export interface DeleteUserWarningProps {
+  /** called when end user cancels account deletion */
+  onCancel: () => void;
+  /** called when user acknowledges account deletion */
+  onConfirmDelete: () => void;
```

Comment:
Maybe `onConfirm` since its the actual action? And can it be optional?

Reply:
+1, https://github.com/aws-amplify/amplify-ui/pull/2927/commits/627877cd862e9c10021e99f72c46d6777bbea02f

Reply:
I don't think it should be optional imo, this is essential to using `Warning` component. This will also block user if customer overrides this incorrectly.